deps/media-playback/media-playback/media-playback.c

```diff
@@ -69,6 +69,23 @@ void media_playback_stop(media_playback_t *mp)
 		mp_media_stop(&mp->media);
 }
 
+void media_playback_set_looping(media_playback_t *mp, bool looping)
+{
+	if (mp->is_cached)
+		mp->cache.looping = looping;
+	else
+		mp->media.looping = looping;
+}
```

Comment:
If this function called after media has already stopped playback, it will not start up playback again, which is what a user would expect if they re-tick the `Loop` option in properties. Instead it will just continue to not play back.

I'm not entirely sure how to deal with this elegantly right now considering playback is in another thread. I think the best thing is just to remove this looping function and restart the media source for the time being until we can come up with a more elegant solution.

Reply:
I applied all other suggestions, but for this one, i didn't drop the looping. Instead I did this https://github.com/obsproject/obs-studio/pull/7455/files#diff-09b0325afe3e08bb7529d6918f3e970513784c791bc496e473c78b2e2519f9a9R499 which I think solves the issue. Please see if it is what you intended. As much as possible, I really don't want to restart the media if only the looping is changed, hence this solution.

Reply:
Sounds good.